### PR TITLE
`Modeling exercises`: Fix missing markers in solution diagram

### LINKED ISSUE #8823: `Modeling`: The solution edit and view in component diagrams show two different diagrams
### Describe the bug

The solution (edit) shows good the lollipop and sockets but the view in the exercise does not show it correctly 

### To Reproduce

1. Create a component diagram 
2. Create the solution
3. Check that all is good in the edition
4. Save it and verify that it is not possible to distinguish the lollipops and sockets. 
5. And also out of the box view 

### Expected behavior

Should show the correct diagram 

### Screenshots

Edit view 
image

Final view 

image

### Which version of Artemis are you seeing the problem on?

7.2.2

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by FelixTJDietrich**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] ~~I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).~~
- [ ] ~~I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).~~
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [ ] ~~I translated all newly inserted strings into English and German.~~

### Motivation and Context

Closes #8823 

Currently the markers in the solution diagram of modeling exercises are not rendered correctly.

### Description

Fix the issue by destroying the hidden Apollon Editor SVG that causes id collisions for the `<marker>` elements with the solution diagram.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Modeling Exercise using a Component Diagram

1. Log in to Artemis
2. Navigate to a course
3. Create a modeling exercise using a Component Diagram
4. Setup a solution containing two components connected via an interface (lollipops), i.e. -o)-
5. Create exercise
6. On the exercise detail view the )-connector to the interface should now be visible. 

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

not changes

### Screenshots

The marker is not showing: (before fix)
image

Marker is showing: (after fix)
image

## Summary by CodeRabbit

- **New Features**
  - Introduced dynamic styling to adjust the top margin of read-only diagrams in the modeling editor based on full-screen mode.
  
- **Style**
  - Added a new CSS class `.readonly-diagram` to enhance the visual representation of read-only diagrams.
  
- **Refactor**
  - Implemented a new method for resource cleanup to prevent SVG `<marker>` id collisions in the modeling editor.

### CODE REVIEW

File: src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.html [hunk 1]
```diff
@@ -108,7 +108,7 @@ <h4 class="modal-title" jhiTranslate="artemisApp.modelingEditor.helpModal.title"
         }
     </div>
     @if (readOnly) {
-        <div class="d-flex align-items-center justify-content-center readonly-diagram" [innerHtml]="readOnlySVG"></div>
+        <div class="d-flex align-items-center justify-content-center readonly-diagram" [innerHtml]="readOnlySVG" [ngStyle]="{ marginTop: isFullScreen ? '30px' : '' }"></div>
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/modeling/shared/modeling-editor.component.html:111]:
```suggestion
        <div class="d-flex align-items-center justify-content-center readonly-diagram" [innerHtml]="readOnlySVG" [style.margin-top]="isFullScreen ? '30px' : ''"></div>
```

**Review by JohannesStoehr — APPROVED** [review]
Code looks good, left one optional suggestion:

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by edkaya — APPROVED** [review]
Tested on ts6, works as expected. )-connector to the interface is visible in exercise detail view. Code also lgtm

**Review by loreanvictor — APPROVED** [review]
code lgtm, also tested on test-server-6 on Safari, Firefox and Chrome and works fine.

**Review by pzdr7 — APPROVED** [review]
Code LGTM 👍 
Works as expected on TS6

**Pull request merged**